src/lustre/lustreAstDependencies.mli

```diff
@@ -16,16 +16,32 @@
 
  *)
 
-(** Graph analysis on Lustre Ast Declarations  
-    Builds a dependency graph of the lustre declarations,
-    to detect circular dependencies and reject them and
-    re-orders node and contract declarations to resolve
-    forward references   
+(** Graph analysis on Lustre Ast Declarations.
+  
+    We build a dependency graph of the lustre declarations
+    to detect circular dependencies and reject them. We also
+    reorder node and contract declarations to resolve
+    forward references and backend cannot handle them.
 
-    @author: Apoorv Ingle *)
+    Note {!Types of dependency analysis}: There are three different kinds of 
+    graph dependency analysis and sorting done here. 
+
+    1. Top level constants and type declarations and functions (starts at [mk_graph_decls]) 
+
+    2. Nodes and contracts (starts at [mk_graph_decls])
+
+    3. Sorting equations of contracts and cirular analysis of node equations 
+
+   TODO: This should module should supercede LustreDependencies when it hardens.     
+
+   @author Apoorv Ingle *)
```

Comment:
I think it would be more clear if you split the `sort_and_check_declarations` function into two functions: one for constant and type declarations, and the other for nodes and contracts. Also, update the reference to `mk_graph_decls` since this is an internal function now.

Reply:
done.